languages/ruby/exercises/concept/booleans/.meta/example.rb

```diff
@@ -0,0 +1,15 @@
+class RidePass
+  MINIMUM_HEIGHT = 120
+
+  def issue(height)
+    rand(100) if height > MINIMUM_HEIGHT
+  end
+
+  def valid?(pass)
+    !pass.nil? && pass != false
```

Comment:
`!pass` can never result in `nil`.

```ruby
def invalid?(pass)
  pass.nil? || ~pass
end
```

But this covers both:

```ruby
def valid?(pass)
  pass
```
Because it is truthy if it is anything other than `nil` or `false` inherent to the Ruby language.

Reply:
Is `!` evaluated before the `.nil?` method in ruby?

Agreed the truth table boils down to pass being the most succinct.  I think you had mentioned last night to not neccessarily have the most idiomatic code in the examples because of students googling and what not, but this way at least if they had found it, they would be able to understand the two concerns in regard to making this pass the test.

Reply:
v2 was in this way for the example code, but anything we present to students should be idiomatic unless showing off something that can be done, or something that is not idiomatic as an example.  At least that is what I have gathered from the V3 discussions.

No, you can't "not" something before it is evaluated.

Reply:
the example, as I understand it, will also not be shown to students.  I would fully expect most to write:

```ruby
def valid?(pass)
  pass
end
```

Reply:
> the example, as I understand it, will also not be shown to students. I would fully expect most to write:

The example will be shown to mentors writing feedback on Representations to provide them with the solution they should try and guide students towards, so it should be the "best" version in line with what is taught in the instructions.

Reply:
I'd personally consider this to be the most idiomatic code for this btw:

```
def valid?(pass)
  !!pass
end
```

I expect `?` methods to return booleans.

Reply:
I expect them to only be able to be used as "booleans" and not necessarily have "Boolean" values returned.  Since the capitalized "B" does not exist in that last word, I suspect that without the double negation it already fits your expectations (Or it was a typographical error.).

The benefit of not having the double negation is that when positive, the value may also be used directly.

```ruby
invalidate_ticket(valid?(pass)) and allow_entry
```

Reply:
```suggestion
    !!pass
```

I think rather than delve to deep on how this value can be reused, as it is contrived for simplicity, it might be easier to have the double-bang

Reply:
Well, not reused, but used.  It will hide a "nil" or "false", it won't be known which it was.  So no way to react accurately.  If it doesn't exist, do we take the route of guiding them to get one?  If it was revoked, do we kick them out of the park?  This way we don't know which it was.